Unable to consume results of system deploy investigation.

This issue is to log and track investigation of `Unable to consume results of system deploy` happening in network with emerging enabled.

**Current findings:** Validation of self created block fails sometimes. Restart persists the error - replaying of the block created fails always. The block can be validated if propose concurrent block on top of the same state (which will be OK) and copy event logs from the new one. NOTE: bad block and new one appear to be identical, the only thing that is not certain is event logs, as it is hard to analyse them. Everything else (what is most important - hashes of the pre state and post state) is identical. You can see this from logs posted in this issue.

**Current guess:** bug in event logs creation.

<!-- Please answer these questions before submitting a bug report. -->

### What did you expect to see?
Node replaying self created block correctly

### What did you see instead?
```
coop.rchain.casper.rholang.types.SystemDeployPlatformFailure$ConsumeFailed$: Platform failure: Unable to consume results of system deploy
        at coop.rchain.casper.rholang.types.SystemDeployPlatformFailure$ConsumeFailed$.<clinit>(SystemDeployUserError.scala)
```

### Steps to reproduce the bug
Run the merging network, after several hours some node might fail replaying self created block.

### What version of RNode are you using?
https://github.com/rchain/rchain/pull/3791

### What is your environment?
Debian, Docker

1. Block creation fails

```
2022-08-16 03:06:20,920 [INFO ] [node-runner-150     ] [c.r.c.b.proposer.BlockCreator]- Block created:
hash: 6fcd3117091491bc656460dc7e2532cce2cef3f255b6885bcc4bca6a92ba00ff
blockNumber: 2743
seqNum: 2738
rejectedDeploys: Set()
rejectedBlocks: Set()
rejectedSenders: Set()
preStateHash: 61c1ee3317f038d1d44d6ff2de3018dee8005391c28ad117fc0a2b0b5621c142
postStateHash: 08e8378647179be24eec2d807a7f4260a48b6c564e899071f7e232cbe2f65c3a
justifications: List(582717, a41cd8, 22c35b, fd28cb, 6d98b0)
bonds: Map(04 -> 50000000000000)
shardId: llbm-testnet}
sender: 048f0998a196fa7694601a94c590dcf101736840b7638a2e8d3c52598dffb47abd94f2adc47dad6035f93148f7022a004da3a6366b39ae4586e9e15800b8436fe0}
sig: 30440220790748c0aa024df4fb850aa63881fe48607c3ea6c575ed920171bc46231bcfb50220020f4d511e32923c97a30f67142188c1d0c18b5ef8266d264c03edc0b7489c7c}
version: 1}
sigAlgorithm: secp256k1}
events: List((3044,882))}
sys events: List(140)}

2022-08-16 03:06:20,921 [INFO ] [node-runner-150     ] [c.r.casper.MultiParentCasper$]- Validating block #2743 6fcd311709... sender: 048f0998a1..., state: 08e8378647..., shard: llbm-testn..., justifi
cations: List(582717efaddf94c6804347cbe073abf5f069a8469f687b0edf4703936915178c, a41cd861b6ab8c7336d740cf3a660d55812b42f588509bb028ba4bdbb198cbd1, 22c35b04fb7694a0253b6673e58ba2bbfeb2816503d186e7e1e8f
5f468791ca1, fd28cb46fe1ccfb81d1bfee2428d6443a85b917870ee792aa426ca2ae30dd770, 6d98b0aa746ca5e27d0731b33e34abe5b3662de95495021e48c737abe5cbfc6b).
2022-08-16 03:06:22,256 [DEBUG] [node-runner-151     ] [c.r.c.merging.MergeScope$    ]- Merging done. Changes: 16 D, 42 K, 42 J; trie actions (100) computed in 11.531376 ms; actions applied in 37.677856 ms
2022-08-16 03:06:22,256 [INFO ] [node-runner-151     ] [c.r.casper.MultiParentCasper$]- PH Set(22, a4, fd, 58, 6d). Finalized fringe state: 76196083c2..., rejectedDeploys: []. FS:CC 40 7
2022-08-16 03:06:26,569 [DEBUG] [node-runner-105     ] [c.r.c.merging.MergeScope$    ]- Merging done. Changes: 66 D, 192 K, 192 J; trie actions (450) computed in 40.491092 ms; actions applied in 71.400173 ms
2022-08-16 03:06:26,569 [INFO ] [node-runner-105     ] [c.r.casper.MultiParentCasper$]- PH Set(22, a4, fd, 58, 6d). Merging FS:CC 38 32. Fringe state 761960
2022-08-16 03:06:26,570 [INFO ] [node-runner-105     ] [c.r.c.r.InterpreterUtil$     ]- Computed parents post state 61c1ee for block #2743 6fcd311709... by 048f0998a1..., fringe rejections: [], rejec
tions: []
2022-08-16 03:06:27,163 [INFO ] [node-runner-158     ] [c.r.c.r.s.RuntimeReplayOps   ]- Evaluating deploy 30440220605718f74775de0c1ed8e93b5b24ae44752c1ae25dc32180402ed32809d4a14802203dff196e29968a3aa
3a8d77d29b8f6c56ed5a4459bdaa5218a22344b6a418a7d
2022-08-16 03:06:27,194 [INFO ] [node-runner-102     ] [c.r.c.r.s.RuntimeReplayOps   ]- after replayDeploys
2022-08-16 03:06:27,197 [INFO ] [node-runner-102     ] [c.r.c.r.InterpreterUtil$     ]- after replayComputeState with Right(<ByteString@7d686d6a size=32 contents="\000\'u\0233\017\344{w\302|2\t\002\3
20\206\274\341\rjs\340i\253\032k\2054\026J\236v">)
2022-08-16 03:06:27,197 [INFO ] [node-runner-102     ] [c.r.c.r.InterpreterUtil$     ]- handleErrors(block.postStateHash, replayResult)
2022-08-16 03:06:27,542 [WARN ] [node-runner-147     ] [c.r.node.runtime.NodeRuntime ]- Graceful shutdown, all resources are successfully closed.
2022-08-16 03:06:27,544 [ERROR] [main                ] [coop.rchain.node.Main$       ]- Unhandled exception in thread main
coop.rchain.casper.rholang.types.SystemDeployPlatformFailure$ConsumeFailed$: Platform failure: Unable to consume results of system deploy
        at coop.rchain.casper.rholang.types.SystemDeployPlatformFailure$ConsumeFailed$.<clinit>(SystemDeployUserError.scala)
```

2. After restart the same block read from block store (6fcd3117091491bc656460dc7e2532cce2cef3f255b6885bcc4bca6a92ba00ff) and attempted to be replayed

```
2022-08-16 05:52:53,637 [INFO ] [node-runner-114     ] [c.r.n.i.ProposerInstance$    ]- Propose started
2022-08-16 05:52:53,643 [INFO ] [node-runner-111     ] [c.r.casper.api.BlockApiImpl  ]- Propose started (seqNum 2738)
2022-08-16 05:52:53,647 [DEBUG] [node-runner-114     ] [c.r.s.LmdbStoreManagerImpl   ]- Creating LMDB environment: /var/lib/rnode/blockstorage
2022-08-16 05:52:53,647 [DEBUG] [node-runner-114     ] [c.r.s.LmdbStoreManagerImpl   ]- Creating LMDB database: blocks, env: /var/lib/rnode/blockstorage
2022-08-16 05:52:53,714 [INFO ] [node-runner-114     ] [c.r.c.b.proposer.Proposer$   ]- Loading 6fcd3117091491bc656460dc7e2532cce2cef3f255b6885bcc4bca6a92ba00ff
2022-08-16 05:52:53,740 [INFO ] [node-runner-114     ] [c.r.casper.MultiParentCasper$]- Validating block #2743 6fcd311709... sender: 048f0998a1..., state: 08e8378647..., shard: llbm-testn..., justifications: List(a41cd861b6ab8c7336d740cf3a660d55812b42f588509bb028ba4bdbb198cbd1, fd28cb46fe1ccfb81d1bfee2428d6443a85b917870ee792aa426ca2ae30dd770, 22c35b04fb7694a0253b6673e58ba2bbfeb2816503d186e7e1e8f5f468791ca1, 582717efaddf94c6804347cbe073abf5f069a8469f687b0edf4703936915178c, 6d98b0aa746ca5e27d0731b33e34abe5b3662de95495021e48c737abe5cbfc6b).
2022-08-16 05:52:59,844 [DEBUG] [node-runner-118     ] [c.r.c.merging.MergeScope$    ]- Merging done. Changes: 16 D, 42 K, 42 J; trie actions (100) computed in 24.172717 ms; actions applied in 39.473324 ms
2022-08-16 05:52:59,846 [INFO ] [node-runner-118     ] [c.r.casper.MultiParentCasper$]- PH Set(22, a4, fd, 58, 6d). Finalized fringe state: 76196083c2..., rejectedDeploys: []. FS:CC 40 7
2022-08-16 05:53:04,027 [DEBUG] [node-runner-112     ] [c.r.c.merging.MergeScope$    ]- Merging done. Changes: 66 D, 192 K, 192 J; trie actions (450) computed in 45.199517 ms; actions applied in 66.751136 ms
2022-08-16 05:53:04,029 [INFO ] [node-runner-112     ] [c.r.casper.MultiParentCasper$]- PH Set(22, a4, fd, 58, 6d). Merging FS:CC 38 32. Fringe state 761960
2022-08-16 05:53:04,032 [INFO ] [node-runner-112     ] [c.r.c.r.InterpreterUtil$     ]- Computed parents post state 61c1ee for block #2743 6fcd311709... by 048f0998a1..., fringe rejections: [], rejections: []
2022-08-16 05:53:04,033 [INFO ] [node-runner-112     ] [c.r.c.r.InterpreterUtil$     ]- randomGenerator
2022-08-16 05:53:04,033 [INFO ] [node-runner-112     ] [c.r.c.r.InterpreterUtil$     ]- replayBlock(incomingPreStateHash, block, rand)
2022-08-16 05:53:04,035 [INFO ] [node-runner-112     ] [c.r.c.r.InterpreterUtil$     ]- before replayComputeState
2022-08-16 05:53:04,101 [INFO ] [node-runner-176     ] [c.r.c.r.s.RuntimeReplayOps   ]- before replayDeploys
2022-08-16 05:53:04,887 [INFO ] [node-runner-175     ] [c.r.c.r.s.RuntimeReplayOps   ]- Evaluating deploy 30440220605718f74775de0c1ed8e93b5b24ae44752c1ae25dc32180402ed32809d4a14802203dff196e29968a3aa3a8d77d29b8f6c56ed5a4459bdaa5218a22344b6a418a7d
2022-08-16 05:53:05,163 [WARN ] [node-runner-118     ] [c.r.node.runtime.NodeRuntime ]- Graceful shutdown, all resources are successfully closed.
2022-08-16 05:53:05,163 [ERROR] [main                ] [coop.rchain.node.Main$       ]- Unhandled exception in thread main
coop.rchain.casper.rholang.types.SystemDeployPlatformFailure$ConsumeFailed$: Platform failure: Unable to consume results of system deploy
	at coop.rchain.casper.rholang.types.SystemDeployPlatformFailure$ConsumeFailed$.<clinit>(SystemDeployUserError.scala)
	at coop.rchain.casper.rholang.syntax.RuntimeOps$.$anonfun$evalSystemDeploy$extension$1(RuntimeSyntax.scala:443)
```

RESULT: failure all the time (attempted 3 times)

--- a/casper/src/main/scala/coop/rchain/casper/merging/MergeScope.scala
+++ b/casper/src/main/scala/coop/rchain/casper/merging/MergeScope.scala
@@ -75,7 +75,7 @@ object MergeScope {
       genesisOpt.map(_.id)
     }
 
-    (MergeScope(fScopeIds, cScopeIds), baseMsg)
+    (MergeScope(fScopeIds, cScopeIds -- baseMsg.toSet), baseMsg)
   }
 
   def merge[F[_]: Concurrent: Log](

